robotframework-ls/src/robotframework_ls/impl/flow_explorer_model_builder.py

```diff
@@ -37,29 +83,61 @@ def build_flow_explorer_model(completion_contexts: List[ICompletionContext]) ->
             }
             suites.append(suite)
 
-            for test_node_info in ast_utils.iter_tests(ast):
-                test_name = f"{suite_name}.{test_node_info.node.name}"
+            for test in ast_utils.iter_tests(ast):
+                test_name = f"{test.node.name} ({suite_name.lower()})"
                 test_body: list = []
-                test = {
+                test_info = {
                     "type": "task",
                     "name": test_name,
                     "doc": "",
                     "setup": None,
                     "teardown": None,
                     "body": test_body,
                 }
-                tasks.append(test)
-                for node_info in ast_utils.iter_all_nodes(
-                    test_node_info.node, recursive=False
-                ):
-                    _build_hierarchy(
-                        completion_context,
-                        node_info.stack,
-                        node_info.node,
-                        suite_name,
-                        test_body,
-                        {},
-                    )
+                tasks.append(test_info)
+                for node_info in ast_utils.iter_all_nodes(test.node, recursive=False):
+                    with recursion_stack.scoped(test_name):
+                        _build_hierarchy(
+                            completion_context=completion_context,
+                            curr_stack=node_info.stack,
+                            curr_ast=node_info.node,
+                            suite_name=suite_name,
+                            parent_body=test_body,
+                            memo={},
+                            recursion_stack=recursion_stack,
+                            user_keywords_collector=user_keywords_collector,
+                        )
+            for user_keyword in ast_utils.iter_keywords(ast):
+                user_keyword_name = f"{user_keyword.node.name} ({suite_name.lower()})"
+                user_keyword_body: list = []
+                user_keyword_info = {
+                    "type": "user-keyword",
+                    "name": user_keyword_name,
+                    "doc": "",
+                    "body": user_keyword_body,
+                }
+
+                # Keywords var will be populated when building hierarchy if importing statements
+                # Checking to see if it already exists before appending
+                for keyw in keywords:
```

Comment:
I think you wanted to do the following here: 

```
if keyw['name'] in user_keywords_collector:
    break
else:
    user_keywords_collector.append(user_keyword_info)
    ...
```

(if you change the `has` for `__contains__`).